### PR TITLE
Add geometry validation documentation

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 1]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
 
    QGIS Server tab in vector layers properties dialog
 
+.. _digitizingmenu:
+
+Digitzing Properties
```
Comment (Gustry) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
Digitizing Properties
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 2]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+====================
+
+|digitizing| The :guilabel:`Digitizing` tab gives access to options that help the
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
 |digitizing| The :guilabel:`Digitizing` tab gives access to options that help to ensure the
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 3]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+quality of digitizied geometries.
+
+Automatic Fixes
+---------------
+
+Options in the :guilabel:`Automatic Fixes` section will directly affect the vertices
+of any geometry which is added or modified.
+If the :guilabel:`Remove duplicate nodes` option is checked, any two subsequent
+vertices with the exactly same coordinate will be removed.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
 vertices with exactly the same coordinates will be removed.
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 4]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+If the :guilabel:`Geometry precision` is set, all vertices will be rounded to
+the closest multiple of the configured geometry precision. The rounding will
+happen in the layer coordiante reference system. Z and M values are not
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
 happen in the layer coordinate reference system. Z and M values are not
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 5]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+rounded. With many map tools, a grid is shown on the canvas while digitizing.
+
+Geometry Checks
+---------------
+
+In the :guilabel:`Geometry checks` section, additional validations on a per
+geometry basis can be activated. Failures in these checks are reported immediately
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
 geometry basis can be activated. Failures in these checks are reported to the user immediately 
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 6]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+the closest multiple of the configured geometry precision. The rounding will
+happen in the layer coordiante reference system. Z and M values are not
+rounded. With many map tools, a grid is shown on the canvas while digitizing.
+
+Geometry Checks
+---------------
+
+In the :guilabel:`Geometry checks` section, additional validations on a per
+geometry basis can be activated. Failures in these checks are reported immediately 
+after any geometry modification to the user in the geometry validation panel.
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
  in the geometry validation panel after any geometry modification.
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 7]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+As long as a check is failing, it is not possible to save the layer.
+The `Is valid` check will run basic validity checks like self intersection on
+geometries.
+
+Topology Checks
+---------------
+
+In the :guilabel:`Topology checks` section, additional topology validation
+checks can be activated. Topology checks will be executed, when the user
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
 checks can be activated. Topology checks will be executed when the user 
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 8]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+As long as a check is failing, it is not possible to save the layer.
+The `Is valid` check will run basic validity checks like self intersection on
+geometries.
+
+Topology Checks
+---------------
+
+In the :guilabel:`Topology checks` section, additional topology validation
+checks can be activated. Topology checks will be executed, when the user 
+saves the layer. Check errors will be reported to the geometry validation
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
 saves the layer. Check errors will be reported in the geometry validation
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 9]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+panel. As long as validation errors are present, the layer can not be saved.
+Topology checks are executed in the area of the modified features. Since other,
+already existing features may be present in the same area, topological errors
+on these features are reported as well and not strictly only errors introduced
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
 concerning these features are reported as well and not strictly only errors introduced
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 1 above)
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
 Digitizing Properties
```

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 6 above)
Comment (m-kuhn) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: I think "immediately after any geometry modification" should be kept together, but the "to the user" part can be moved before. What do you think?

**Comment by m-kuhn** [comment]
Thanks a lot for the review @havatv very good inputs.
Any comment on the remaining question?

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 6 above)
Comment (havatv) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: I agree that they should be kept together.  What about
`Immediately after any geometry modification, failures in these checks are reported to the user in the geometry validation panel.`

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 10]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+
+Topology Checks
+---------------
+
+In the :guilabel:`Topology checks` section, additional topology validation
+checks can be activated. Topology checks will be executed when the user 
+saves the layer. Check errors will be reported in the geometry validation
+panel. As long as validation errors are present, the layer can not be saved.
+Topology checks are executed in the area of the modified features. Since other,
+already existing features may be present in the same area, topological errors
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: Trying to understand "present in the same area": do you mean current map canvas extent or feature extent?
what about sthg like: "Topological errors are reported for both the existing features (in the current map canvas?) and the new (/or edited?) features (in the current session)."

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 11]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+concerning these features are reported as well as errors introduced
+in the current edit session.
+
+The :guilabel:`Gap check` will check for gaps between neighbouring polygons.
+
+|gapcheck|
```
Comment (DelazJ) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]: I wonder if we should keep three separate images or merge them in a single one with a caption.

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 6 above)
Comment (m-kuhn) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: Much better, thanks

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 10 above)
Comment (m-kuhn) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: "The bounding box of the features modified or added in the current edit session"

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 11 above)
Comment (m-kuhn) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: I thought by having them separated, they can appear exactly where they need. What's the advantage of a single monolithic image?

File: source/docs/user_manual/working_with_vector/vector_properties.rst [hunk 12]
```diff
@@ -3284,6 +3284,63 @@ format of the image. Currently png, jpg and jpeg image formats are supported.
...
+
+Topology Checks
+---------------
+
+In the :guilabel:`Topology checks` section, additional topology validation
+checks can be activated. Topology checks will be executed when the user 
+saves the layer. Check errors will be reported in the geometry validation
+panel. As long as validation errors are present, the layer can not be saved.
+Topology checks are executed in the area of the bounding box of the modified
+features. Since other already existing features may be present in the same
```
Comment (havatv) [inline source/docs/user_manual/working_with_vector/vector_properties.rst]:
```suggestion
 features. Since other features may be present in that
```

Comment (havatv) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: I think that "already existing" is not necessary here.  The end of the sentence makes it even more clear what we are talking about.

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 11 above)
Comment (havatv) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: Separate images are OK. Easier to maintain?

File: source/docs/user_manual/working_with_vector/vector_properties.rst (hunk 12 above)
Comment (m-kuhn) [reply source/docs/user_manual/working_with_vector/vector_properties.rst]: better?

**Comment by m-kuhn** [comment]
If anyone is eager to merge, I think it's good to go ;)

**Comment by havatv** [comment]
Thanks, @m-kuhn !

**Pull request merged**